src/lib/conditions.ts

```diff
@@ -0,0 +1,59 @@
+type TCondition = (() => boolean) | boolean;
+
+function isAFunction(func: TCondition): func is Callback {
+	return typeOf(func) === "function";
+}
+
+export class X {
```

Comment:
Should all condition methods/names be UPPERCASE?

Reply:
That's an amazing suggestion to keep with the standard, most of the times you find them being written in UPPERCASE in books. 

Will make this change.